crypto/dilithium/pqcrystals_dilithium_ref_common/sign.c

```diff
@@ -69,6 +69,19 @@ int crypto_sign_keypair_internal(ml_dsa_params *params,
   /* FIPS 204: line 9 Compute H(rho, t1) and line 10 write secret key */
   shake256(tr, TRBYTES, pk, params->public_key_bytes);
   pack_sk(params, sk, rho, tr, key, &t0, &s1, &s2);
+
+  /* FIPS 204. Section 3.6.3 Destruction of intermediate values. */
+  OPENSSL_cleanse(seedbuf, sizeof(seedbuf));
+  OPENSSL_cleanse(tr, sizeof(tr));
+  OPENSSL_cleanse(&rho, SEEDBYTES);
```

Comment:
This still doesn't seem right. You're passing in the address of rho (which is pointer type whose size is sizeof(uint_8*) and clearing SEEDBYTES from that address. I believe this is a buffer overrun on the stack frame. Can someone check my work? I think what you want is to either not do any cleansing at all and assume that the pointer values are not sensitive, or you want to replace SEEDBYTES with sizeof(rho) to basically set the pointer to 0. Same goes for the other pointers here.

Reply:
Check what we did for ML-KEM for the same requirement, I was following this same methodology: https://github.com/aws/aws-lc/pull/1883